overlord/snapstate/snapstate.go

```diff
@@ -1048,36 +1048,50 @@ func updateManyFiltered(ctx context.Context, st *state.State, names []string, us
 		return nil, nil, err
 	}
 
-	refreshOpts := &store.RefreshOptions{IsAutoRefresh: flags.IsAutoRefresh}
-	updates, stateByInstanceName, ignoreValidation, err := refreshCandidates(ctx, st, names, user, refreshOpts)
-	if err != nil {
-		return nil, nil, err
-	}
+	var toUpdate []update
+	var stateByInstanceName map[string]*SnapState
+	if flags.IsAutoRefresh {
```

Comment:
this change seems premature, but maybe your TODO refers to that, also I'm not sure we can implement it this way, do we hold continuously the state lock from the place where wound set refresh-hints to here?  anyway it feels too magic, I think we want to carry refresh-hints as value on top of sticking them in the state. We probably need to reorganize the functions a bit differently to avoid this magic.